File: SupplierPortal.Web/Modules/Market/Supplier/SupplierForm.cs
using Serenity.ComponentModel;

namespace SupplierPortal.Market.Forms;

[FormScript("Market.Supplier")]
[BasedOnRow(typeof(SupplierRow), CheckNames = true)]
public class SupplierForm
{
    public string CompanyName { get; set; }
    public string Email { get; set; }
    public string OfficalName { get; set; }
    public string OfficalSurname { get; set; }
    public string CallNumber { get; set; }
}

File: SupplierPortal.Web/Modules/ServerTypes/Market/SupplierForm.ts
﻿import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SupplierForm {
    CompanyName: StringEditor;
    Email: StringEditor;
    OfficalName: StringEditor;
    OfficalSurname: StringEditor;
    CallNumber: StringEditor;
}

export class SupplierForm extends PrefixedContext {
    static readonly formKey = 'Market.Supplier';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SupplierForm.init)  {
            SupplierForm.init = true;

            var w0 = StringEditor;

            initFormType(SupplierForm, [
                'CompanyName', w0,
                'Email', w0,
                'OfficalName', w0,
                'OfficalSurname', w0,
                'CallNumber', w0
            ]);
        }
    }
}